types/react-instantsearch-dom/index.d.ts

```diff
@@ -144,5 +120,43 @@ export class SortBy extends React.Component<any> {}
 /**
  * The Stats component displays the total number of matching hits and the time it took to get them (time spent in the Algolia server).
  */
-export class Stats extends React.Component<{translations?: { [key: string]: (n: number, ms: number) => string }}> {}
+export class Stats extends React.Component<{
+  translations?: {
+    [key: string]: (n: number, ms: number) => string;
+  };
+}> {}
 export class ToggleRefinement extends React.Component<any> {}
+
+export class VoiceSearch extends React.Component<any> {}
+
+export class QueryRuleCustomData extends React.Component<any> {}
+
+// helper functions
+export function getInsightsAnonymousUserToken(): string | undefined;
+export function createClassNames(baseName: string): (...elements: string[]) => string | string[];
+
+export interface VoiceSearchHelperParams {
+  searchAsYouSpeak: boolean;
+  language?: string;
+  onQueryChange: (query: string) => void;
+  onStateChange: () => void;
+}
+
+export type Status = 'initial' | 'askingPermission' | 'waiting' | 'recognizing' | 'finished' | 'error';
+
+export interface VoiceListeningState {
+  status: Status;
+  transcript: string;
+  isSpeechFinal: boolean;
+  errorCode?: any;
```

Comment:
Would be nice to have this enumerated.  To allow for future error codes you can do something like:

`errorCode: 'Bad Request' | 'Not Found' | string`

This effectively becomes a string type then, but at least the current list of values is documented so consumers can write a switch statement around it.

Reply:
It's actually  `type SpeechRecognitionErrorCode = "no-speech" | "aborted" | "audio-capture" | "network" | "not-allowed" | "service-not-allowed" | "bad-grammar" | "language-not-supported";` from lib/dom (of typescript), but I wasn't sure if it should be included here. When we publish the types from our our source file when the rest is converted to TS that should automatically be fixed.

Reply:
```suggestion
  errorCode?: "no-speech" | "aborted" | "audio-capture" | "network" | "not-allowed" | "service-not-allowed" | "bad-grammar" | "language-not-supported";
```